### ISSUE TITLE
Metalama does not support (global) `using static` & `using alias` directives

### ISSUE THREAD

**Issue by Cheatoid** [postsharp/Metalama#19]
## Steps to reproduce
This bug can also be reproduced at [try.metalama.net](https://try.metalama.net).

### Source Code
```cs
static class Program
{
    static void Main()
    {
        try
        {
            Add(1, 1);
            Add(0, 1);
        }
        catch { }
    }

    [Log]
    static int Add(int a, int b)
    {
        if (a == 0)
        {
            throw new ArgumentOutOfRangeException(nameof(a));
        }
        return a + b;
    }
}
```

### Aspect Code
```cs
#define BUG // to prove the bug; remove this line if you want to have successful compilation

global using global::System;
global using global::Metalama.Framework.Aspects;

#if BUG
// Metalama does not recognize 'using alias' directive as C# does.
global using SEXCEPT = global::System.Exception;
global using OMA = Metalama.Framework.Aspects.OverrideMethodAspect;
#endif

#if BUG
// Metalama does not recognize 'using static' directive as C# does.
global using static global::System.Console;
#else
using static global::System.Console;
#endif

public class LogAttribute :
#if BUG
    OMA
#else
    OverrideMethodAspect
#endif
{
    public override dynamic? OverrideMethod()
    {
        WriteLine(meta.Target.Method.ToDisplayString() + ": started.");
        try
        {
            var result = meta.Proceed();
            WriteLine(meta.Target.Method.ToDisplayString() + ": succeeded.");
            return result;
        }
#if BUG
        catch (SEXCEPT ex)
#else
        catch (Exception ex)
#endif
        {
            WriteLine(meta.Target.Method.ToDisplayString() + ": failed due " + ex.Message);
            throw;
        }
    }
}
```

## Expected behavior
The above example should compile with `#define BUG` line, just like it compiles without it.

## Actual behavior
Program output:
```
The compile-time project could not be compiled. In most cases, this is due to a problem in your code and can be diagnosed using the other reported errors. If, however, you believe this is due to a bug in Metalama, please report the issue and include diagnostic information available in 'C:\local\Temp\Metalama\CompileTimeTroubleshooting\0.5.22.0-9877888c\5f35bfa6-de98-445e-8f36-bd002be6db6d'.
Aspect.cs(3,1): error CS0138: A 'using namespace' directive can only be applied to namespaces; 'Exception' is a type not a namespace. Consider a 'using static' directive instead
Aspect.cs(3,1): error CS0138: A 'using namespace' directive can only be applied to namespaces; 'OverrideMethodAspect' is a type not a namespace. Consider a 'using static' directive instead
Aspect.cs(3,1): error CS0138: A 'using namespace' directive can only be applied to namespaces; 'Console' is a type not a namespace. Consider a 'using static' directive instead
```

## References
[C# `global using`](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-10#global-using-directives)
[C# `using alias`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-directive#:~:text=type%20with%20a-,using%20alias%20directive,-.)

**Comment by prochan2** [comment]
Thank you for reporting this issue. We've filed this bug and we'll let you know as soon as it's fixed.

**Comment by aweneer** [comment]
Hello @Cheatoid, this bug should be now fixed in our newest released version of Metalama 0.5.32-preview.
